docs/modules/clients/pages/java.adoc

```diff
@@ -1950,6 +1899,64 @@ You can also configure it within the Spring context, as shown below:
 </beans>
 ----
 
+== Handling Failures
+
+There are three main failure cases and configurations you can perform to achieve proper behavior.
+
+**Handling Client Connection Failure:**
+
+While the client is trying to connect initially to one of the members in the
+`ClientNetworkConfig.addressList`, all the members might be not available.
+Instead of giving up, throwing an exception and stopping the client,
+the client retries to connect as configured which is described in the
+<<configuring-client-connection-retry, Configuring Client Connection Retry section>>.
+
+The client executes each operation through the already established connection to the cluster.
+If this connection(s) disconnects or drops, the client tries to reconnect as configured.
+
+**Handling Unresponsive Connections**
+
+It is possible for a connection to be open, but unresponsive, due to several scenarios like JVM pauses or TCP half-open sockets. If this occurs, a client which already has an open connection to that member will *not* be able to tell that the member is unresponsive because the connection is still open and no errors are received from the member side. Client operations will stay in-flight, potentially indefinitely, in this scenario. To avoid this, Hazelcast provides <<java-client-failure-detectors, Client Failure Detectors>> to mark members which appear to be unresponsive or down. Once the member is marked as suspect by the client, operations will fail as expected.

```

Comment:
The last part might be confusing. `operations will fail as expected` , operation will not always fail, it might be retried depending on the timeout. We might mention it and direct them to next section for details.

Reply:
That is a good point. I have also found during tests that operations attempted against a "suspect" member by the client can fall victim to the member-side property `hazelcast.operation.call.timeout.millis` as their timeout instead. I believe this happens when clients call a responsive node, which then attempts to contact an unresponsive node which has the primary copy of the data requested. That would be a member-to-member timeout, and would fit with what we describe at the bottom of [Invocation Lifecycle](https://docs.hazelcast.com/imdg/4.2/consistency-and-replication/invocation-lifecycle) documentation.

Does that sound correct?

Reply:
I have updated the wording and removed references to failure, as like you said it may or may not fail. I also ultimately decided against adding any reference to the operation call timeout, as the use case I already described in which a user might encounter it is not common and is already covered in theory by the Invocation Lifecycle documentation.